tests/e2e/test_pv_scale_ocs_create_delete_pvcs.py

```diff
@@ -0,0 +1,338 @@
+                else:
+                    future_cephfs_rwx_obj.append(
+                        executor.submit(helpers.create_multiple_pvcs,
+                                        sc_name=cephfs_sc_obj.name, namespace=self.namespace,
+                                        number_of_pvc=number_of_pvc, access_mode=mode,
+                                        size=size))
+        cephfs_rwo_pvcs = [pvc_obj.result() for pvc_obj in future_cephfs_rwo_obj]
+        cephfs_rwo_pvcs = flatten(cephfs_rwo_pvcs)
+        cephfs_rwx_pvcs = [pvc_obj.result() for pvc_obj in future_cephfs_rwx_obj]
+        cephfs_rwx_pvcs = flatten(cephfs_rwx_pvcs)
+        temp_pvc_objs.extend(cephfs_rwo_pvcs)
+        temp_pvc_objs.extend(cephfs_rwx_pvcs)
+        with ThreadPoolExecutor() as executor:
+            for mode in access_modes:
+                if mode == constants.ACCESS_MODE_RWO:
+                    future_rbd_rwo_obj.append(
+                        executor.submit(helpers.create_multiple_pvcs,
+                                        sc_name=cephfs_sc_obj.name, namespace=self.namespace,
+                                        number_of_pvc=number_of_pvc, access_mode=mode,
+                                        size=size))
+                else:
+                    future_rbd_rwx_obj.append(
+                        executor.submit(helpers.create_multiple_pvcs,
+                                        sc_name=rbd_sc_obj.name, namespace=self.namespace,
+                                        number_of_pvc=number_of_pvc, access_mode=mode,
+                                        size=size))
+        rbd_rwo_pvcs = [pvc_obj.result() for pvc_obj in future_rbd_rwo_obj]
+        rbd_rwo_pvcs = flatten(rbd_rwo_pvcs)
+        rbd_rwx_pvcs = [pvc_obj.result() for pvc_obj in future_rbd_rwx_obj]
+        rbd_rwx_pvcs = flatten(rbd_rwx_pvcs)
+        temp_pvc_objs.extend(rbd_rwo_pvcs)
+        temp_pvc_objs.extend(rbd_rwx_pvcs)
+        # Appending all the pvc obj to base case param for cleanup and evaluation
+        self.all_pvc_obj.extend(temp_pvc_objs)
+        # Creating pods with respective pvc created above
+        # PODs are create for CephFS RWX & RWO and RBD RWO pvcs
+        log.info(f"created {len(temp_pvc_objs)} pvc_objects")
+        with ThreadPoolExecutor() as executor:
+            for pvc_obj in cephfs_rwo_pvcs:
+                pod_name = helpers.create_unique_resource_name(
+                    resource_description=f"{pvc_obj.name}-pod",
+                    resource_type="pod")
+                future_temp_pod_objs.append(
+                    executor.submit(helpers.create_pod,
+                                    interface_type=constants.CEPHFS_INTERFACE,
+                                    pvc_name=pvc_obj.name, do_reload=False,
+                                    namespace=self.namespace))
+                log.info(f"Created pod {pod_name} using pvc {pvc_obj.name}")
+            for pvc_obj in cephfs_rwx_pvcs:
```

Comment:
overall question/feedback/comment:
I see you creating different lists for PVC from different access_modes and intefaces (same for pods), and later on combine the lists.
Why do you need this separation?

Reply:
The reason for using separate list is to use different pvc objects, this TC has all type of pvc, also it will be helpful to create different kind of pods,
soon will provide a patch using functions, which will remove lot of lists.

Reply:
@RazTamir updated the code with helper functions, please take a look.